### ISSUE TITLE
Problem with curl 8.7.1

### ISSUE THREAD

**Issue by OengusOiw** [tj/n#799]
# Upgrade with n - problem curl.

## Configuration Details

```bash
Problem :

sudo n latest
response : curl: (23) Failed writing received data to disk/application
Error: failed to download version index (https://nodejs.org/dist/index.tab)

$ n doctor

COMMAND LOCATIONS AND VERSIONS

bash
/usr/bin/bash
GNU bash, version 5.2.21(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2022 Free Software Foundation, Inc.
Licence GPLv3+ : GNU GPL version 3 ou ultérieure <http://gnu.org/licenses/gpl.html>

Ceci est un logiciel libre ; vous être libre de le modifier et de le redistribuer.
AUCUNE GARANTIE n'est fournie, dans les limites permises par la loi.

n
/usr/local/bin/n
v9.2.1

node
/usr/local/bin/node
v21.7.1
JavaScript engine: v8

npm
/usr/local/bin/npm
10.5.0

tar
/usr/bin/tar
tar (GNU tar) 1.35
Copyright © 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Écrit par John Gilmore et Jay Fenlason.

curl or wget
/usr/bin/curl
curl 8.7.1 (x86_64-pc-linux-gnu) libcurl/8.7.1 OpenSSL/3.1.5 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.7 libpsl/0.21.2 libssh2/1.11.0 nghttp2/1.60.0 librtmp/2.3 OpenLDAP/2.5.16
Release-Date: 2024-03-27, security patched: 8.7.1-1
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

uname
Linux HP-5CD112TWVG 6.7.9-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.7.9-2 (2024-03-13) x86_64 GNU/Linux


SETTINGS

n
node mirror: https://nodejs.org/dist
node downloads mirror: https://nodejs.org/download
install destination: /usr/local
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
ls-remote max matches: 20

Proxy


CHECKS

Checking n install destination is in PATH...
good

Checking n install destination priority in PATH...
good

Checking npm install destination...
good

Checking prefix folders...
good

Checking permissions for cache folder...
good

Checking permissions for install folders...
good

Checking mirror is reachable...
good

$ command -v node

/usr/local/bin/node

$ node -p process.platform

linux

```

**Comment by shadowspawn** [comment]
There is an issue with curl 8.7.1 which fails with the `--compressed` flag. First reported here in #798, and the related curl issue found thanks to [p-d-boiko](https://github.com/p-d-boiko):

- https://github.com/curl/curl/issues/13209

**Comment by luckman212** [comment]
I came to report the same. Glad it's been identified. I tried to follow the paper trail of related issues but got a bit lost. It seems there have been fixed pushed but it will take some days/weeks for everything to catch up in the package repos.

**Comment by shadowspawn** [comment]
The [fix](https://github.com/curl/curl/issues/13209) is listed on the [pending release page](https://curl.se/dev/release-notes.html) to go public on 2024-05-22, so might be a while before it is available. 😢   

The quick manual hack is to edit the `n` shell script and remove the use of `--compressed` in `do_get_index`.

Would people who are being affected by the problem like a release of `n` which removes `--compressed` until it is fixed in curl?

**Comment by shadowspawn** [comment]
I saw curl 8.7.1 is available in Homebrew, so getting into common distributions. Released n 9.2.3 which removes use of the `--uncompressed` option for now (#801).
